lib/OpenQA/WebAPI/Controller/API/V1/Bug.pm

```diff
@@ -116,6 +117,64 @@ sub show {
 
 =over 4
 
+=item cleanup()
+
+Delete all bugs that are not referenced from any job comments anymore.
+
+=back
+
+=cut
+
+sub cleanup {
+    my ($self) = @_;
+
+    my $bugrefs = $self->bugrefcount();
+    my $bugs    = $self->schema->resultset("Bugs")->search();
+    my %toclean;
+    for my $bug ($bugs->all) {
+        if (!defined $bugrefs->{$bug->bugid}) {
+            $toclean{$bug->id} = $bug->bugid;
+            $self->emit_event('openqa_bug_delete', {id => $bug->id, bugid => $bug->bugid});
+            $bug->delete;
+        }
+    }
+
+    $self->render(json => {bugs => \%toclean});
+}
+
+=over 4
+
+=item bugrefcount()
+
+Return a hashref of all bugs referenced by job comments with the number of references.
+
+=back
+
+=cut
+
+
+sub bugrefcount {
+    my ($self) = @_;
+
+    my $comments = $self->schema->resultset("Comments")->search({-not => {'job_id' => undef}});
```

Comment:
Don't quote `job_id` here and remove the useless spaces in the rest of the function.